### PR TITLE
Add missing exports @testing library/react

### PR DESCRIPTION
**Pull request by samtsai**
- Links to documentation: https://testing-library.com/docs/dom-testing-library/api-queries#screen
- Link to GitHub or NPM:
[`screen`](https://github.com/testing-library/dom-testing-library/blob/master/src/screen.js)
see [Typescript `screen.d.ts`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/testing-library__dom/screen.d.ts)
[`waitForElementToBeRemoved`](https://testing-library.com/docs/dom-testing-library/api-async#waitforelementtoberemoved)
see [Typescript `wait-for-element-to-be-removed.d.ts`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/testing-library__dom/wait-for-element-to-be-removed.d.ts)
- Type of contribution: addition | fix

Other notes:
Backported fixes to previous versions for accuracy and passing tests.

### CODE REVIEW

File: definitions/npm/@testing-library/react_v10.x.x/flow_v0.104.x-/react_v10.x.x.js [hunk 1]
```diff
@@ -202,13 +202,31 @@ declare module '@testing-library/react' {
     eventProperties?: TInit
   ) => boolean;
 
+  declare type Screen<Queries = GetsAndQueries> = {
+    ...Queries,
```
Comment (pascalduez) [inline definitions/npm/@testing-library/react_v10.x.x/flow_v0.104.x-/react_v10.x.x.js:206]: Spreading in an inexact object doesn't behave a intended.

File: definitions/npm/@testing-library/react_v10.x.x/flow_v0.104.x-/react_v10.x.x.js [hunk 2]
```diff
@@ -254,18 +272,8 @@ declare module '@testing-library/react' {
...
-      container?: HTMLElement,
-      timeout?: number,
-      interval?: number,
-      mutationObserverOptions?: MutationObserverInit,
-    |}
-  ): Promise<HTMLElement>;
-
   declare export function waitForElementToBeRemoved<T>(
-    callback?: () => T,
+    callback: (() => T) | T,
```
Comment (pascalduez) [inline definitions/npm/@testing-library/react_v10.x.x/flow_v0.104.x-/react_v10.x.x.js:276]: Humm okay, so if you pass an HTML Element as `callback` it's properly inferred? 
https://testing-library.com/docs/dom-testing-library/api-async#waitforelementtoberemoved

File: definitions/npm/@testing-library/react_v10.x.x/flow_v0.104.x-/test_react_v10.x.x.js [hunk 3]
```diff
@@ -104,8 +105,6 @@ describe('waitForElement', () => {
 
 describe('waitForElementToBeRemoved', () => {
   it('should fail on invalid inputs', () => {
-    // $ExpectError
-    waitForElementToBeRemoved(1);
```
Comment (pascalduez) [inline definitions/npm/@testing-library/react_v10.x.x/flow_v0.104.x-/test_react_v10.x.x.js:108]: That's the problem, it's now too loose and let you use anything.

Comment (samtsai) [reply definitions/npm/@testing-library/react_v10.x.x/flow_v0.104.x-/test_react_v10.x.x.js:108]: I don't disagree, what's an alternative way to type this?

File: definitions/npm/@testing-library/react_v10.x.x/flow_v0.104.x-/react_v10.x.x.js (hunk 1 above)
Comment (samtsai) [reply definitions/npm/@testing-library/react_v10.x.x/flow_v0.104.x-/react_v10.x.x.js:206]: What would be the more appropriate approach? I was trying to follow previous discussions and carry over the latest

File: definitions/npm/@testing-library/react_v10.x.x/flow_v0.104.x-/test_react_v10.x.x.js (hunk 3 above)
Comment (pascalduez) [reply definitions/npm/@testing-library/react_v10.x.x/flow_v0.104.x-/test_react_v10.x.x.js:108]: Like it was before, two separate declarations.

File: definitions/npm/@testing-library/react_v10.x.x/flow_v0.104.x-/react_v10.x.x.js (hunk 1 above)
Comment (pascalduez) [reply definitions/npm/@testing-library/react_v10.x.x/flow_v0.104.x-/react_v10.x.x.js:206]: Try to use exact as much as possible, if not possible intersection?

**Pull request merged**